feat: Add open in editor button with configurable editor preference

VAIBHAVSING opened the pull request:
## Summary
This PR implements the ability to open a session's working directory in an external editor directly from CodeLayer.

## Changes
- Added an 'Open in Editor' button (📝 icon) next to the Archive button in the session detail view
- Added a settings section to configure the preferred editor (Cursor, VS Code, or Zed)
- Implemented Cmd+Shift+E (⌘+Shift+E on Mac, Ctrl+Shift+E on Windows/Linux) hotkey to open the working directory
- Used tauri-plugin-opener for cross-platform editor launching

## Implementation Details
- Added new preferences in `src/lib/preferences.ts` for storing editor choice
- Updated `ActionButtons` component to include the new button
- Updated `ResponseInput` component to handle the editor opening logic and hotkey
- Added editor selection dropdown in `SettingsDialog`

## Testing
- Ran existing tests - all previously passing tests still pass
- Ran linter - no new warnings introduced
- TypeScript compilation successful for our changes

Closes #479
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds 'Open in Editor' feature with configurable editor preference and hotkey support in CodeLayer.
> 
>   - **Behavior**:
>     - Adds 'Open in Editor' button in `ActionButtons` and `ActiveSessionActionButtons` components.
>     - Supports editors: Cursor, VS Code, Zed, and system default.
>     - Implements hotkey Cmd+Shift+E for opening the editor.
>   - **Preferences**:
>     - Adds `PREFERRED_EDITOR_KEY` in `preferences.ts` for storing preferred editor.
>     - Provides editor options in `EDITOR_OPTIONS`.
>     - Updates `SettingsDialog` to include editor selection dropdown.
>   - **Commands**:
>     - Adds `open_in_editor` command in `lib.rs` to handle directory opening logic.
>     - Uses `tauri-plugin-opener` for cross-platform support.
>   - **Testing**:
>     - Existing tests pass, no new warnings from linter.
>     - TypeScript compilation successful.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=humanlayer%2Fhumanlayer&utm_source=github&utm_medium=referral)<sup> for e9081595ec46713161915dfd1c75fb77f6215d85. You can [customize](https://app.ellipsis.dev/humanlayer/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

VAIBHAVSING committed:
feat: Add open in editor button with configurable editor preference

- Add 'Open in Editor' button next to Archive button in session detail
- Support Cursor, VS Code, and Zed editors
- Add editor preference selector in Settings dialog
- Implement Cmd+Shift+E hotkey to open working directory in preferred editor
- Use tauri-plugin-opener for cross-platform editor launching

Resolves #479

VAIBHAVSING committed:
style: Format code with prettier

VAIBHAVSING committed:
feat : added open in IDE feature to pwd to an editor

VAIBHAVSING committed:
feat :format

VAIBHAVSING committed:
feat: refractor

VAIBHAVSING commented:
@dexhorthy  PTAL 
hopefully its working on mac 
now it working on  ubantu

VAIBHAVSING commented:
<img width="1854" height="1048" alt="Screenshot from 2025-10-18 14-15-09" src="https://github.com/user-attachments/assets/55fa5f1c-7721-407f-8b32-ef6821ec1348" />

K-Mistele reviewed (changes requested):
The sub-text is hard to read here when a given option is focused - it should probably change to be more visible like how most of our buttons do

Also, the "Cursor" text in the select component's selection appears to have a weird left indent, and that box could use a bit more padding as well 
<img width="737" height="428" alt="Screenshot 2025-10-23 at 1 57 08 PM" src="https://github.com/user-attachments/assets/1fdc1334-6bec-4257-b15d-789126125b37" />

left a few other comments too

VAIBHAVSING committed:
refactor: use tauri-plugin-opener for editor launch

VAIBHAVSING committed:
feat: change default editor to VS Code

VAIBHAVSING committed:
fix: use unicode shift character in keyboard shortcut

VAIBHAVSING committed:
fix: add proper hotkey scope and contentEditable support

VAIBHAVSING commented:
## Implementation Update for PR #730

I've addressed the feedback from the PR review with the following key improvements:

### ✅ Changes Implemented

1. **Switched to `tauri-plugin-opener` for cross-platform compatibility**
   - Removed custom shell command logic (400+ lines)
   - Now uses native Tauri opener API for reliable editor launching
   - Works on macOS without requiring CLI tools to be installed
   - Platform-specific app name mapping (e.g., "Visual Studio Code" on macOS, "Code" on Windows)

2. **Changed default editor from Cursor to VS Code**
   - Updated default in preferences
   - VS Code is more commonly installed across platforms

3. **Fixed keyboard shortcut display**
   - Changed from `⌘+Shift+E` to `⌘⇧E` to match codebase style
   - Now consistent with other keyboard shortcuts in the app

4. **Added proper hotkey scope configuration**
   - Added `HOTKEY_SCOPES.SESSION_DETAIL` and `SESSION_DETAIL_ARCHIVED` scopes
   - Added `enableOnContentEditable: true` for proper editor integration
   - Hotkey now properly scoped to session detail views

### 📝 Commits
- `7f04001` - refactor: use tauri-plugin-opener for editor launch
- `d91d221` - feat: change default editor to VS Code
- `b7bf598` - fix: use unicode shift character in keyboard shortcut  
- `857218f` - fix: add proper hotkey scope and contentEditable support

### 🔄 Remaining Work

The **Settings UI improvements** are not yet implemented. This requires:
- Making the editor selector match the model selector button style
- Improving focus states for better readability  
- Fixing padding and indentation issues

I plan to address the UI improvements in a follow-up commit after getting feedback on the core functionality changes.

### 🧪 Testing Needed

Please test on macOS to verify that editors open correctly without CLI tools installed. The new implementation should work with just the applications installed in `/Applications`.

cc @dexhorthy @K-Mistele - ready for review!

VAIBHAVSING committed:
Merge main and fix conflicts

- Resolved conflicts in lib.rs by keeping both read_last_log_lines and save_window_state functions
- Resolved conflicts in SettingsDialog.tsx by merging editor preferences with log handling
- Combined imports and state management from both branches
- Applied UI improvements: better padding, improved focus states for select items

VAIBHAVSING committed:
fix: improve Settings UI styling per PR feedback

- Add px-2.5 padding to SelectTrigger to fix weird left indent
- Change description text from text-muted-foreground to opacity-70 with group-data state
- Make sub-text more readable when focused by increasing opacity on focus
- Update keyboard shortcut display from ⌘+Shift+E to ⌘⇧E
- Add group class to SelectItem for better state management

Addresses feedback from @K-Mistele in PR #730
